addons/sale_pdf_quote_builder/models/sale_order.py

```diff
@@ -25,18 +25,39 @@ class SaleOrder(models.Model):
         string="Customizable PDF Form Fields",
         readonly=False,
     )
+    ignore_default_ids = fields.Many2many(
+        string="Default Quotation Document To Ignore",
+        relation='ignore_default_relation',
+        comodel_name='quotation.document',
+        readonly=False,
+    )
+
+    # === CRUD METHODS === #
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        orders = super().create(vals_list)
+        default_quotes = self.env['quotation.document'].search([('default', '=', True)])
+        for order in self:
+            order.quotation_document_ids |= default_quotes & order.available_product_document_ids
+            order.ignore_default_ids = order.available_product_document_ids - order.quotation_document_ids
+        return orders
 
     # === COMPUTE METHODS === #
 
     @api.depends('sale_order_template_id')
     def _compute_available_product_document_ids(self):
+        default_quotes = self.env['quotation.document'].search([('default', '=', True)])
         for order in self:
             order.available_product_document_ids = self.env['quotation.document'].search(
                 [], order='sequence'
             ).filtered(lambda doc:
                 self.sale_order_template_id in doc.quotation_template_ids
                 or not doc.quotation_template_ids
             )
+            order.quotation_document_ids &= order.available_product_document_ids
```

Comment:
This line is unrelated to the task, right? It makes sense IMO, but I'm not sure this is the best place to put it. Indeed, if the available documents change, then the quotation's documents could be updated without warning (even if the quotation was already sent). Maybe you could perform this validation in a constraint instead?

Reply:
Yes, its unrelated. I had the issue a couple times where adding a quotation template to a sale order would increase the set of available document, and then removing the quotation template would decrease the set of available document without removing the non available document from the quote, if I added any. 

Im not familiar enough with constraints so I might not have understood your idea. What you mean is adding a method that will clear the unavailable document from the quote every time the `available_...` field is changed? Should I warn the user that some document have been removed (if any) or just update the `quotation_document_ids` or block the change until he removed all the documents related to the template?